lib/src/data_table_2.dart

```diff
@@ -400,25 +409,56 @@ class DataTable2 extends DataTable {
     return contents;
   }
 
-  Widget _buildHeadingCell(
-      {required BuildContext context,
-      required EdgeInsetsGeometry padding,
-      required Widget label,
-      required String? tooltip,
-      required bool numeric,
-      required VoidCallback? onSort,
-      required bool sorted,
-      required bool ascending,
-      required double effectiveHeadingRowHeight,
-      required MaterialStateProperty<Color?>? overlayColor}) {
+  Widget _buildHeadingCell({
+    required BuildContext context,
+    required EdgeInsetsGeometry padding,
+    required Widget label,
+    required String? tooltip,
+    required bool numeric,
+    required VoidCallback? onSort,
+    required bool sorted,
+    required bool ascending,
+    required double effectiveHeadingRowHeight,
+    required MaterialStateProperty<Color?>? overlayColor,
+    AlignmentDirectional? alignment,
+  }) {
     final ThemeData themeData = Theme.of(context);
 
     var customArrows =
         sortArrowBuilder != null ? sortArrowBuilder!(ascending, sorted) : null;
+
+    final currentDirection = Directionality.of(context);
+    final bool isRTL = currentDirection == TextDirection.rtl;
+
+    final alignedLabel = switch (alignment) {
```

Comment:
Why add `alignedLabel` whilst there's already `label = Container()`, which has alignment set there and it depends on the type of the column (numeric or not)